### DISCUSSION TITLE
Font shaped incorrectly

### DISCUSSION THREAD

**Discussion by photopea** [harfbuzz/harfbuzz#4690] [General] [answered]
I have the font Amiri - [amiri.zip](https://github.com/harfbuzz/harfbuzz/files/15148486/amiri.zip). 

I ask Harfbuzz to shape the string "08.05.2024" and it returns incorrect glyphs for dots. I use left-to-right direction, no features, and not calling "hb_buffer_guess_segment_properties". Adding a latin character (e.g. "a") to the end of string fixes it. Could you try to shape it yourself? Why does it return these glyphs for dots?

**Comment by behdad** [comment, marked as answer]
Looks like the Arabic dot. I suppose in the absence of any script information, Amiri defaults to Arabic. @khaledhosny

**Comment by photopea** [comment]
When I use this font in Adobe Photoshop, the dot "defaults" to a latin dot. Is Photoshop ignoring the Opentype standard? Or does Photoshop provide some extra information to a shaping engine, which I am not providing to Harfbuzz?

**Reply by behdad** [reply]
> Or does Photoshop provide some extra information to a shaping engine, which I am not providing to Harfbuzz?

If you pass `--script=latn` you get the Latin dots. Perhaps Photoshop is doing that instead of using the `DFLT` script system.

**Reply by khaledhosny** [reply]
Amiri’s default period glyph is designed for Arabic and the glyph suitable for Latin is substituted with `locl` feature under `latn` script.

Photoshop seems to set the script based on the language setting, if I copy the string above and set the language to English or any other language that uses the Latin script, I get the rounded dot, if I set it to Arabic I get the default lozenge dot.

**Reply by khaledhosny** [reply]
If I add a Latin letter to the string, then it uses rounded dot. Which suggests it guesses script from the language only if it can’t detect it from the text itself.
